lib/controllers/createController.js

```diff
@@ -3,58 +3,53 @@
 var path = require('path');
 
 module.exports = function (apiPlatformService, BPromise, fileSystemRepository,
-  localService, logger, messages, userOrganizationService, workspaceRepository) {
+    localService, logger, messages, userOrganizationService, workspaceRepository) {
+  var createAPIStrategy = {
+    create: function (newApi) {
+      logger.info(messages.creatingAPI());
+      return apiPlatformService.createAPI(
+        newApi.bizGroup,
+        newApi.name,
+        newApi.versionName,
+        newApi.rootRamlPath);
+    },
+    getAPIIdentifier: function (apis, newApi, parametersStrategy) {
+      return parametersStrategy.getAPIName(apis)
+        .tap(function (apiName) {
+          newApi.name = apiName;
+        });
+    }
+  };
+
+  var createAPIVersionStrategy = {
+    create: function (newApi) {
+      logger.info(messages.creatingAPIVersion());
+      return apiPlatformService.createAPIVersion(
+        newApi.bizGroup,
+        newApi.id,
+        newApi.versionName,
+        newApi.rootRamlPath);
+    },
+    getAPIIdentifier: function (apis, newApi, parametersStrategy) {
+      return parametersStrategy.getAPI(apis)
+        .tap(function (api) {
+          newApi.id = api.id;
+        });
+    }
+  };
+
```

Comment:
In order to keep the return of the module on the top of the file, move this objects to createAPIorAPIVersion